src/components/block/Block.ts

```diff
@@ -0,0 +1,278 @@
+    if (response) {
+      this.eventBus().emit(Block.EVENTS.FLOW_RENDER);
+    }
+  }
+
+  // Может переопределять пользователь, необязательно трогать
+  componentDidUpdate(oldProps: Props, newProps: Props) {
+    // добавить сравнение пропсов
+    if (oldProps && newProps) {
+      return true;
+    }
+    return true;
+  }
+
+  setProps = (nextProps: object) => {
+    if (!nextProps) {
+      return;
+    }
+
+    this._removeEvents();
+
+    Object.assign(this.props, nextProps);
+    this.eventBus().emit(Block.EVENTS.FLOW_CDU);
+  };
+
+  get element() {
+    return this._element;
+  }
+
+  _render() {
+    const block = this.render();
+
+    const newElement = block.firstElementChild as HTMLElement;
+
+    if (this._element && newElement) {
+      this._element.innerHTML = '';
+      this._element.replaceWith(newElement);
+
+      this._addEvents();
+    }
+
+    this._element = newElement;
+    this._addEvents();
+    // Этот небезопасный метод для упрощения логики
+    // Используйте шаблонизатор из npm или напишите свой безопасный
+    // Нужно не в строку компилировать (или делать это правильно),
+    // либо сразу в DOM-элементы возвращать из compile DOM-ноду
+  }
+
+  // Может переопределять пользователь, необязательно трогать
+  render(): DocumentFragment {
+    return new DocumentFragment();
+  }
+
+  getContent() {
+    return this._element;
+  }
+
+  compile(template: string, props: Props) {
+    const propsWithStubs: Props = {...props};
+
+    Object.entries(this.children).forEach(([key, child]) => {
+      if (Array.isArray(child)) {
+        propsWithStubs[key] = child
+          .map((childItem) => `<div data-id="${childItem._id}"></div>`)
+          .join(' ');
+      } else {
+        propsWithStubs[key] = `<div data-id="${child._id}"></div>`;
+      }
+    });
+
+    const fragment = this._createDocumentElement('template') as HTMLTemplateElement;
+
+    fragment.innerHTML = new Templator(template).compile(propsWithStubs);
+
+    Object.values(this.children).forEach((child) => {
+      if (Array.isArray(child)) {
+        child.forEach((childItem) => {
+          const stub = fragment.content.querySelector(`[data-id="${childItem._id}"]`);
+          stub?.replaceWith(childItem.getContent() as Node);
+        });
+      } else {
+        const stub = fragment.content.querySelector(`[data-id="${child._id}"]`);
+        stub?.replaceWith(child.getContent() as Node);
+      }
+    });
+
+    return fragment.content;
+  }
+
+  _makePropsProxy(props: Props) {
+    // Можно и так передать this
+    // Такой способ больше не применяется с приходом ES6+
+    // const self = this;
+
+    const eventBus = this.eventBus.bind(this);
+
+    return new Proxy(props, {
+      set(target, prop: string, value) {
+        // this здесь указывает на объект handler
+        const _target = getDeepCopy(props);
+        if (prop in target) {
+          target[prop] = value;
+          eventBus().emit(Block.EVENTS.FLOW_CDU, _target, props);
```

Comment:
А можно использовать стрелочные функции, тогда вообще не придется обращаться к замыканию
```suggestion
    return new Proxy(props, {
      set: (target, prop: string, value) => {
        // this здесь указывает на объект handler
        const _target = getDeepCopy(props);
        if (prop in target) {
          target[prop] = value;
          this.eventBus().emit(Block.EVENTS.FLOW_CDU, _target, props);
```

Reply:
исправлено в 3 спринте